cpp/daal/src/algorithms/dtrees/forest/df_train_dense_default_impl.i

```diff
@@ -633,6 +748,188 @@ typename DataHelper::NodeType::Base * TrainBatchTaskBase<algorithmFPType, DataHe
+
+        base.nLeft        = split.nLeft;
+        base.improvement  = improvement;
+        base.impurityLeft = split.left;
+        _helper.convertLeftImpToRight(n, curImpurity, split);
+        base.impurityRight = split.left;
+        if (!(base.node = makeSplit(iFeature, split.featureValue, split.featureUnordered, nullptr, nullptr, curImpurity.var))) return nullptr;
+        base.isLeaf = false;
+        base.featureUnordered |= bool(split.featureUnordered);
+        base.node->count = n;
+
+        --remainingSplitNodes;
+        s = binaryHeap.push(base);
+        if (!s.ok()) return nullptr;
+    }
+    else
+        return makeLeaf(_aSample.get() + iStart, n, curImpurity, nClasses);
+
+    while (!binaryHeap.empty())
+    {
+        WorkItem & src = binaryHeap.pop();
+        if (src.isLeaf) continue; // this node is leaf
+
+        // create leftChild
+        typename DataHelper::TSplitData splitLeft;
+        IndexType iFeatureLeft;
+        WorkItem leftChild(src.featureUnordered, src.start, src.nLeft, src.level + 1);
+
+        if (!remainingSplitNodes || terminateCriteria(src.nLeft, src.level + 1, src.impurityLeft))
+            src.node->kid[0] = makeLeaf(_aSample.get() + src.start, src.nLeft, src.impurityLeft, nClasses);
+        else if (findBestSplit(src.start, src.nLeft, src.impurityLeft, iFeatureLeft, splitLeft))
+        {
+            const size_t nLeft   = splitLeft.nLeft;
+            const double imp     = src.impurityLeft.var;
+            const double impLeft = splitLeft.left.var;
+
+            // check impurity decrease
+            double improvement = imp * src.nLeft - impLeft * nLeft - (src.nLeft - nLeft) * (imp - impLeft);
+            if (improvement < _minImpurityDecrease)
+                src.node->kid[0] = makeLeaf(_aSample.get() + src.start, src.nLeft, src.impurityLeft, nClasses);
+            else
+            {
+                if (_par.varImportance == training::MDI) addImpurityDecrease(iFeatureLeft, src.nLeft, src.impurityLeft, splitLeft);
+
+                leftChild.nLeft        = splitLeft.nLeft;
+                leftChild.improvement  = improvement;
+                leftChild.impurityLeft = splitLeft.left;
+                _helper.convertLeftImpToRight(src.nLeft, src.impurityLeft, splitLeft);
+                leftChild.impurityRight = splitLeft.left;
+
+                if (!(leftChild.node =
+                          makeSplit(iFeatureLeft, splitLeft.featureValue, splitLeft.featureUnordered, nullptr, nullptr, src.impurityLeft.var)))
+                    return nullptr;
+
+                src.node->kid[0] = leftChild.node;
+                leftChild.isLeaf = false;
+                leftChild.featureUnordered |= bool(splitLeft.featureUnordered);
+                leftChild.node->count = src.nLeft;
+                --remainingSplitNodes;
+            }
+        }
+        else
+            src.node->kid[0] = makeLeaf(_aSample.get() + src.start, src.nLeft, src.impurityLeft, nClasses);
```

Comment:
minnor: always use brackets (code style)
```suggestion
        {
            src.node->kid[0] = makeLeaf(_aSample.get() + src.start, src.nLeft, src.impurityLeft, nClasses);
        }
```

Reply:
it was fixed